cassdegrees/api/degreeRules.py

```diff
@@ -1,52 +1,161 @@
-class degreeRules:
-    JSONString = []
+import json
 
-    def getRule(self, ruleType, match):
-        rule = {"type": ruleType, "match": match}
+class DegreeRules:
+    """
+    A class which represents degree rules as a k-ary tree and stores them in a list.
 
-        return rule
+    Rules should be created using the rule_ functions, which return a dictionary
+    representing the rule. Rules can then be arranged into an expression and parsed
+    through the class to build their k-ary tree representation.
 
-    """ 
-        ["(", A, "OR", "B", ")", "AND", "(", "C", "OR", "D", ")"]
+    Calling get_rules_as_json will return the built expression as a JSON serialised string
+    that can be stored in the database under [degree].[rules].
     """
-    def buildExpression(self, expressionList):
-        tree = {}
+
+    def __init__(self, expression_list):
+        self.expressionTree = []
+        self.build_expression(expression_list)
+
+    def __init__(self):
+        self.expressionTree = []
+
+    def build_expression(self, expression_list):
+        """
+        Accepts a list of tokens that represent a boolean expression of degree rules
+        and converts them into a k-ary tree of lists and dictionaries that can be
+        stored in JSON format.
+
+        :param expression_list: (list) a list of tokens in the format of a boolean expression.
+                                A token must be one of:
+                                    "(", ")", "AND", "OR", rule
```

Comment:
Just noting that it looks like this function requires the external tokens to be '(' and ')' as it will not work without them